### ISSUE TITLE
Clean up documentation about encoding dense vectors

### ISSUE THREAD

**Issue by lintool** [castorini/pyserini#1063]
@MXueguang @crystina-z 

I just noticed we have two doc pages that have a lot of overlap...

https://github.com/castorini/pyserini/blob/master/docs/usage-encode.md
https://github.com/castorini/pyserini/blob/master/docs/usage-dense-indexes.md

Why do we have both? Should we collapse?

@crystina-z iirc, you were the person to last use this feature with mr.tydi?
@MXueguang do we want to make this consistent with tevatron?

**Comment by crystina-z** [comment]
true I was using `python -m pyserini.encode`.

So `encode` and `index` are the same concept in pyserini? and we now just keep one of the page (also rename the title into index/encode passage)?

**Comment by lintool** [comment]
The way @MXueguang and I are thinking about it - and of course, open to suggestions - is that "encode" is generating the doc vectors, whereas "index" would be something like building HNSW indexes.

Of course, if we "encode" into Faiss flat index, that "encoding" could also serve as an index...

@crystina-z thoughts?

**Comment by MXueguang** [comment]
> I just noticed we have two doc pages that have a lot of overlap...
yes for the two docs here, we want to merge them as doc for "encoding". we don't have an index doc for dense rn.

**Comment by crystina-z** [comment]
@lintool if we feel like articulate the difference between "encode" and "index", maybe we can merge the current [use-dense-indexes.md](https://github.com/castorini/pyserini/blob/master/docs/usage-dense-indexes.md) into [usage-encode](https://github.com/castorini/pyserini/blob/master/docs/usage-encode.md
), and put a specific index example (e.g. prep hnsw from encoded vectors) into the use-dense-indexes.md instead?

also mentioning "for faiss flat, the encoded vectors can be used as index already" etc.

**Comment by lintool** [comment]
yup, agreed. @crystina-z would you mind taking an initial stab at it?

**Comment by lintool** [comment]
hey @crystina-z I'll reorganize the docs better first via #1070 and then you can do clean-up?

**Comment by crystina-z** [comment]
oh sure, I'll clean the doc based on #1070
